File: Components/DevExpressVCL/Demos/ExpressPivotGrid/Delphi/OrderReportsDemo/OrderReportsMain.pas
unit OrderReportsMain;

{$I cxVer.inc}

interface

uses
  Windows, Messages, Variants, Forms, Dialogs,
  SysUtils, Menus, StdCtrls, Controls, cxLookAndFeels, DemoBasicMain,
  cxControls, cxCustomPivotGrid, cxDBPivotGrid, DemoBasicDM, cxStyles,
  cxCustomData, cxGraphics, cxFilter, cxData, cxDataStorage, cxEdit, DB, cxDBData,
  Classes, cxClasses;

type
  TfrmOrderReport = class(TfrmDemoBasicMain)
    DBPivotGrid: TcxDBPivotGrid;
    pgfPurchaseDate: TcxDBPivotGridField;
    pgfPaymentType: TcxDBPivotGridField;
    pgfQuantity: TcxDBPivotGridField;
    pgfCarName: TcxDBPivotGridField;
    pgfUnitPrice: TcxDBPivotGridField;
    pgfCompanyName: TcxDBPivotGridField;
    pgfPaymentAmount: TcxDBPivotGridField;
    procedure FormCreate(Sender: TObject);
    procedure pgfPaymentTypeGetGroupImageIndex(Sender: TcxPivotGridField;
      const AItem: TcxPivotGridViewDataItem; var AImageIndex: Integer;
      var AImageAlignHorz: TAlignment;
      var AImageAlignVert: TcxAlignmentVert);
  private
  protected
    function GetPivotGrid: TcxCustomPivotGrid; override;
  public
    { Public declarations }
  end;

var
  frmOrderReport: TfrmOrderReport;

implementation

{$R *.dfm}

function TfrmOrderReport.GetPivotGrid: TcxCustomPivotGrid;
begin
  Result := DBPivotGrid;
end;

procedure TfrmOrderReport.FormCreate(Sender: TObject);
begin
  inherited;
  PivotGrid.ApplyBestFit;
end;

procedure TfrmOrderReport.pgfPaymentTypeGetGroupImageIndex(
  Sender: TcxPivotGridField; const AItem: TcxPivotGridViewDataItem;
  var AImageIndex: Integer; var AImageAlignHorz: TAlignment;
  var AImageAlignVert: TcxAlignmentVert);
var
  Card: string;
begin
  Card := VarToStr(AItem.Value);
  if SameText(Card, 'AmEx') then
    AImageIndex := 1
  else
    if SameText(Card, 'Cash') then
      AImageIndex := 0
    else
      if SameText(Card, 'Master') then
        AImageIndex := 2
      else
        if SameText(Card, 'Visa') then
          AImageIndex := 3;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressPivotGrid/CBuilder/OrderReportsDemo/OrderReportsMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "OrderReportsMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxLookAndFeels"
#pragma link "DemoBasicMain"
#pragma link "cxControls"
#pragma link "cxCustomPivotGrid"
#pragma link "cxDBPivotGrid"
#pragma resource "*.dfm"
TfrmOrderReport *frmOrderReport;
//---------------------------------------------------------------------------
__fastcall TfrmOrderReport::TfrmOrderReport(TComponent* Owner)
        : TfrmDemoBasicMain(Owner)
{
}

void __fastcall TfrmOrderReport::FormCreate(TObject* Sender)
{
  TfrmDemoBasicMain::FormCreate(Sender); 
  PivotGrid()->ApplyBestFit();
}

TcxCustomPivotGrid* __fastcall TfrmOrderReport::PivotGrid()
{
  return DBPivotGrid;
}

//---------------------------------------------------------------------------
void __fastcall TfrmOrderReport::pgfPaymentTypeGetGroupImageIndex(
      TcxPivotGridField *Sender, const TcxPivotGridViewDataItem *AItem,
      int &AImageIndex, TAlignment &AImageAlignHorz,
      TcxAlignmentVert &AImageAlignVert)
{
  AnsiString Card = ((TcxPivotGridViewDataItem *)AItem)->Value;
  if (SameText(Card, "Cash")) AImageIndex = 0;
  else if (SameText(Card, "AmEx")) AImageIndex = 1;
       else if (SameText(Card, "Master")) AImageIndex = 2;
            else if (SameText(Card, "Visa")) AImageIndex = 3;
}
//---------------------------------------------------------------------------

